lib/services/downloads_service_backend.dart

```diff
@@ -0,0 +1,1515 @@
+import 'dart:async';
+import 'dart:convert';
+import 'dart:core';
+import 'dart:io';
+
+import 'package:background_downloader/background_downloader.dart';
+import 'package:collection/collection.dart';
+import 'package:finamp/components/global_snackbar.dart';
+import 'package:finamp/services/downloads_service.dart';
+import 'package:flutter/scheduler.dart';
+import 'package:get_it/get_it.dart';
+import 'package:isar/isar.dart';
+import 'package:logging/logging.dart';
+import 'package:path/path.dart' as path_helper;
+
+import '../models/finamp_models.dart';
+import '../models/jellyfin_models.dart';
+import 'finamp_settings_helper.dart';
+import 'finamp_user_helper.dart';
+import 'jellyfin_api_helper.dart';
+
+part 'isar_downloads_backend.g.dart';
+
+class IsarPersistentStorage implements PersistentStorage {
+  final _isar = GetIt.instance<Isar>();
+
+  @override
+  Future<void> storeTaskRecord(TaskRecord record) =>
+      _store(IsarTaskDataType.taskRecord, record.taskId, record);
+
+  @override
+  Future<TaskRecord?> retrieveTaskRecord(String taskId) =>
+      _get(IsarTaskDataType.taskRecord, taskId);
+
+  @override
+  Future<List<TaskRecord>> retrieveAllTaskRecords() =>
+      _getAll(IsarTaskDataType.taskRecord);
+
+  @override
+  Future<void> removeTaskRecord(String? taskId) =>
+      _remove(IsarTaskDataType.taskRecord, taskId);
+
+  @override
+  Future<void> storePausedTask(Task task) =>
+      _store(IsarTaskDataType.pausedTask, task.taskId, task);
+
+  @override
+  Future<Task?> retrievePausedTask(String taskId) =>
+      _get(IsarTaskDataType.pausedTask, taskId);
+
+  @override
+  Future<List<Task>> retrieveAllPausedTasks() =>
+      _getAll(IsarTaskDataType.pausedTask);
+
+  @override
+  Future<void> removePausedTask(String? taskId) =>
+      _remove(IsarTaskDataType.pausedTask, taskId);
+
+  @override
+  Future<void> storeResumeData(ResumeData resumeData) =>
+      _store(IsarTaskDataType.resumeData, resumeData.taskId, resumeData);
+
+  @override
+  Future<ResumeData?> retrieveResumeData(String taskId) =>
+      _get(IsarTaskDataType.resumeData, taskId);
+
+  @override
+  Future<List<ResumeData>> retrieveAllResumeData() =>
+      _getAll(IsarTaskDataType.resumeData);
+
+  @override
+  Future<void> removeResumeData(String? taskId) =>
+      _remove(IsarTaskDataType.resumeData, taskId);
+
+  @override
+  (String, int) get currentDatabaseVersion => ("FinampIsar", 1);
+
+  @override
+  // This should come from finamp settings if migration needed
```

Comment:
Basically, if we want to migrate to a new schema we should store the current version in finamp settings.  Right now that is not stored at all, because there is only 1 version so we can just assume what version we have.

Reply:
But what is the benefit of adding it to the settings? That it stays in the original version, even if we update the default? Or something else?

Reply:
Putting it in settings allows us to know whether we need to migrate or not, the same as all the other migration flags we already have.  It's not there now because we never need to migrate, because there's only one version at the moment.

Reply:
Ah okay, because the settings are available during startup.